frame/nfts/src/impl_nonfungibles.rs

```diff
@@ -0,0 +1,230 @@
+		} else {
+			let key = BoundedSlice::<_, _>::try_from(key).ok()?;
+			Attribute::<T, I>::get((collection, Some(item), namespace, key)).map(|a| a.0.into())
+		}
+	}
+
+	/// Returns the attribute value of `item` of `collection` corresponding to `key`.
+	///
+	/// When `key` is empty, we return the item metadata value.
+	///
+	/// By default this is `None`; no attributes are defined.
+	fn collection_attribute(collection: &Self::CollectionId, key: &[u8]) -> Option<Vec<u8>> {
+		if key.is_empty() {
+			// We make the empty key map to the item metadata value.
+			CollectionMetadataOf::<T, I>::get(collection).map(|m| m.data.into())
+		} else {
+			let key = BoundedSlice::<_, _>::try_from(key).ok()?;
+			Attribute::<T, I>::get((
+				collection,
+				Option::<T::ItemId>::None,
+				AttributeNamespace::CollectionOwner,
+				key,
+			))
+			.map(|a| a.0.into())
+		}
+	}
+
+	/// Returns `true` if the `item` of `collection` may be transferred.
+	///
+	/// Default implementation is that all items are transferable.
+	fn can_transfer(collection: &Self::CollectionId, item: &Self::ItemId) -> bool {
+		match (
+			CollectionConfigOf::<T, I>::get(collection),
+			ItemConfigOf::<T, I>::get(collection, item),
+		) {
+			(Some(cc), Some(ic))
+				if cc.is_setting_enabled(CollectionSetting::TransferableItems) &&
+					ic.is_setting_enabled(ItemSetting::Transferable) =>
+				true,
+			_ => false,
+		}
+	}
+}
+
+impl<T: Config<I>, I: 'static> Create<<T as SystemConfig>::AccountId, CollectionConfigFor<T, I>>
+	for Pallet<T, I>
+{
+	/// Create a `collection` of nonfungible items to be owned by `who` and managed by `admin`.
+	fn create_collection(
+		who: &T::AccountId,
+		admin: &T::AccountId,
+		config: &CollectionConfigFor<T, I>,
+	) -> Result<T::CollectionId, DispatchError> {
+		// DepositRequired can be disabled by calling the force_create() only
+		ensure!(
+			!config.has_disabled_setting(CollectionSetting::DepositRequired),
+			Error::<T, I>::WrongSetting
+		);
+
+		let collection =
+			NextCollectionId::<T, I>::get().unwrap_or(T::CollectionId::initial_value());
+
+		Self::do_create_collection(
+			collection,
+			who.clone(),
+			admin.clone(),
+			*config,
+			T::CollectionDeposit::get(),
+			Event::Created { collection, creator: who.clone(), owner: admin.clone() },
+		)?;
+		Ok(collection)
+	}
+}
+
+impl<T: Config<I>, I: 'static> Destroy<<T as SystemConfig>::AccountId> for Pallet<T, I> {
+	type DestroyWitness = DestroyWitness;
+
+	fn get_destroy_witness(collection: &Self::CollectionId) -> Option<DestroyWitness> {
+		Collection::<T, I>::get(collection).map(|a| a.destroy_witness())
+	}
+
+	fn destroy(
+		collection: Self::CollectionId,
+		witness: Self::DestroyWitness,
+		maybe_check_owner: Option<T::AccountId>,
+	) -> Result<Self::DestroyWitness, DispatchError> {
+		Self::do_destroy_collection(collection, witness, maybe_check_owner)
+	}
+}
+
+impl<T: Config<I>, I: 'static> Mutate<<T as SystemConfig>::AccountId, ItemConfig> for Pallet<T, I> {
```

Comment:
Hello there! One question I have is why are not the `set_attribute` and `set_typed_attribute` methods implemented for the trait?

You seem to provide the capability to do so in the `do_set_attribute` extrinsic, but if I want to use this pallet as another pallet's dependency I won't be able to set attributes using the nonfungibles trait.

I was wondering if there was any particular reason or limitation for having not implemented them?

Reply:
Hi @DidacSF ! Thanks for pointing this out. I just added the implementation of those methods in this interim PR: https://github.com/paritytech/substrate/pull/12919

Reply:
Hey that's great! Thank you for your work!